packages/features/ee/workflows/api/scheduleEmailReminders.ts

```diff
@@ -40,10 +70,11 @@ async function handler(req: NextApiRequest, res: NextApiResponse) {
   // delete batch_ids with already past scheduled date from scheduled_sends
   const pageSize = 90;
   let pageNumber = 0;
-  const deletePromises: Promise<any>[] = [];
+
+  const remindersToDelete: { referenceId: string | null }[] = [];
 
   while (true) {
-    const remindersToDelete = await prisma.workflowReminder.findMany({
+    const newRemindersToDelete = await prisma.workflowReminder.findMany({
```

Comment:
This is changed everywhere below too. The while loop now exclusively gathers all workflowReminders. Once we've collected all the reminders, we process them in a separate step, generating the necessary promises to handle the required requests.

@hariombalhara That should solve the `unhandled promise rejection`, right?